File: 品味Java的21種設計模式/dp_java/state/example7/LeaveRequestModel.java
package cn.javass.dp.state.example7;
/**
 * 请假单对象
 */
public class LeaveRequestModel {
	/**
	 * 请假人
	 */
	private String user;
	/**
	 * 请假开始时间
	 */
	private String beginDate;
	/**
	 * 请假天数
	 */
	private int leaveDays;
	/**
	 * 审核结果
	 */
	private String result;
	
	public String getResult() {
		return result;
	}

	public void setResult(String result) {
		this.result = result;
	}

	public String getUser() {
		return user;
	}

	public String getBeginDate() {
		return beginDate;
	}

	public int getLeaveDays() {
		return leaveDays;
	}
	public void setUser(String user) {
		this.user = user;
	}

	public void setBeginDate(String beginDate) {
		this.beginDate = beginDate;
	}

	public void setLeaveDays(int leaveDays) {
		this.leaveDays = leaveDays;
	}	
}

File: 品味Java的21種設計模式/dp_cpp/state/example7/LeaveRequestModel.cpp
#include "LeaveRequestModel.h"

namespace cn {
namespace javass {
namespace dp {
namespace state {
namespace example7 {

std::string LeaveRequestModel::getResult() {
    return result;
}

void LeaveRequestModel::setResult(std::string result) {
    this->result = result;
}

std::string LeaveRequestModel::getUser() {
    return user;
}

std::string LeaveRequestModel::getBeginDate() {
    return beginDate;
}

int LeaveRequestModel::getLeaveDays() {
    return leaveDays;
}

void LeaveRequestModel::setUser(std::string user) {
    this->user = user;
}

void LeaveRequestModel::setBeginDate(std::string beginDate) {
    this->beginDate = beginDate;
}

void LeaveRequestModel::setLeaveDays(int leaveDays) {
    this->leaveDays = leaveDays;
}
}
}
}
}
}
